In [0]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

In [0]:
train_df = pd.read_csv("train (1).csv")
test_df = pd.read_csv("test (3).csv")

combine = [train_df, test_df]

## Step 1: Checking the Samples

For the first step, we are going to check the statictics of our samples vs. the statistics of the whole population.  There are really only 2 reasons that the stats. won't be anything close to each other.



1.   We don't have enough data.
2.   Our data is made up or fake.

Both of these reasons are bad things.  If any of these are true, we can't do much with the data.



In [9]:
train_df.describe()

Id   MSSubClass  LotFrontage        LotArea  OverallQual  \
count  1460.000000  1460.000000  1201.000000    1460.000000  1460.000000   
mean    730.500000    56.897260    70.049958   10516.828082     6.099315   
std     421.610009    42.300571    24.284752    9981.264932     1.382997   
min       1.000000    20.000000    21.000000    1300.000000     1.000000   
25%     365.750000    20.000000    59.000000    7553.500000     5.000000   
50%     730.500000    50.000000    69.000000    9478.500000     6.000000   
75%    1095.250000    70.000000    80.000000   11601.500000     7.000000   
max    1460.000000   190.000000   313.000000  215245.000000    10.000000   

       OverallCond    YearBuilt  YearRemodAdd   MasVnrArea   BsmtFinSF1  \
count  1460.000000  1460.000000   1460.000000  1452.000000  1460.000000   
mean      5.575342  1971.267808   1984.865753   103.685262   443.639726   
std       1.112799    30.202904     20.645407   181.066207   456.098091   
min       1.000000  1872.000000   1950.000000     0.000000     0.000000   
25%       5.000000  1954.000000   1967.000000     0.000000     0.000000   
50%       5.000000  1973.000000   1994.000000     0.000000   383.500000   
75%       6.000000  2000.000000   2004.000000   166.000000   712.250000   
max       9.000000  2010.000000   2010.000000  1600.000000  5644.000000   

           ...         WoodDeckSF  OpenPorchSF  EnclosedPorch    3SsnPorch  \
count      ...        1460.000000  1460.000000    1460.000000  1460.000000   
mean       ...          94.244521    46.660274      21.954110     3.409589   
std        ...         125.338794    66.256028      61.119149    29.317331   
min        ...           0.000000     0.000000       0.000000     0.000000   
25%        ...           0.000000     0.000000       0.000000     0.000000   
50%        ...           0.000000    25.000000       0.000000     0.000000   
75%        ...         168.000000    68.000000       0.000000     0.000000   
max        ...         857.000000   547.000000     552.000000   508.000000   

       ScreenPorch     PoolArea       MiscVal       MoSold       YrSold  \
count  1460.000000  1460.000000   1460.000000  1460.000000  1460.000000   
mean     15.060959     2.758904     43.489041     6.321918  2007.815753   
std      55.757415    40.177307    496.123024     2.703626     1.328095   
min       0.000000     0.000000      0.000000     1.000000  2006.000000   
25%       0.000000     0.000000      0.000000     5.000000  2007.000000   
50%       0.000000     0.000000      0.000000     6.000000  2008.000000   
75%       0.000000     0.000000      0.000000     8.000000  2009.000000   
max     480.000000   738.000000  15500.000000    12.000000  2010.000000   

           SalePrice  
count    1460.000000  
mean   180921.195890  
std     79442.502883  
min     34900.000000  
25%    129975.000000  
50%    163000.000000  
75%    214000.000000  
max    755000.000000  

[8 rows x 38 columns]

According to the internet, the mean house price in Iowa is about 175,000.  This isn't far off from our 180,921 that our training sample has.

Now we just have to check the test sample.

This means that in the our predictions should roughly have a mean of about 175,000.  If they are far off, then that may be a bad sign.

In [43]:
train_df['LandSlope'].isnull().any()

False

## Step 2: Understanding Features

**Sale Price**: What we are trying to predict.  This is the price of the house in dollars.

**MSSubClass**: This is the building class.  A categorical integer with no missing values.

**MSZoning**: This is the general zoning classification.  A categorical String with no missing values.

**LotFrontage**: This is the amount of street that is connected to the property.  This is a float value with some missing data.  This could be due to some homes not being connected to any street.  *This may be solved by setting missing values to zero.*

**LotArea**: The sqft of area.  An int value which is a continous feature.  There is no missing values.

**Street**: The type of road access.  Cat. strings with no missing values.  Most of values are paved.  May not be so important.

**Alley**: The type of alley access.  A cat. string with a ton of missing values.  This is due to a house not having any alley.  *Solve by setting nan values to 'None'*.

**LotShape**: This is the est. shape of the property.  This ranges from 'Reg'=Regular, and 3 other irregular shapes.

**LandContour**: This is the slope of the land.  Lvl=Level property, HLS=a hill, LOW=Low Grade, etc.

**Utilities**: The type of utilities that are available.  This may not be important as only a single home does not have utilities.

**LotConfig**: The configuration of the lot.  This is the location of the home on the street.